### PR TITLE
ISSUE-3557: Add rename table command

### LINKED ISSUE #3557: Rename table statement
**Summary**

```
RENAME TABLE old_table TO new_table;
```
https://dev.mysql.com/doc/refman/8.0/en/rename-table.html

### PR DESCRIPTION
**Pull request by kevinw66**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Add rename table command

## Changelog

- New Feature

## Related Issues

Fixes #3557

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
@mergify update

**Review by Junnplus — CHANGES_REQUESTED** [review]
Thanks!

We may need some unit tests for interpreter and parser.

**Comment by kevinw66** [comment]
> Thanks!
> 
> We may need some unit tests for interpreter and parser.

@Junnplus Done

File: common/planners/src/plan_table_rename.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+use std::sync::Arc;
+
+use common_datavalues::DataSchema;
+use common_datavalues::DataSchemaRef;
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq)]
+pub struct RenameTablePlan {
+    pub tenant: String,
+    pub maps: Vec<RenameTableMap>,
```
Comment (Junnplus) [inline common/planners/src/plan_table_rename.rs:23]: `entities` could be better.

**Comment by kevinw66** [comment]
> Hi @kevinw66 ,
> 
> Thanks for the cool feature. Is it possible to add a stateless test for the `rename table`? For example, the `truncate table` stateless test is: https://github.com/datafuselabs/databend/blob/85878247b295863000134ab569205ae7a9247b08/tests/suites/0_stateless/05_ddl/05_0003_ddl_truncate_table.sql

What's the name rule ? I saw two 05_0000 sql file and none 05_0002 sql file

**Comment by BohuTANG** [comment]
> > Hi @kevinw66 ,
> > Thanks for the cool feature. Is it possible to add a stateless test for the `rename table`? For example, the `truncate table` stateless test is: https://github.com/datafuselabs/databend/blob/85878247b295863000134ab569205ae7a9247b08/tests/suites/0_stateless/05_ddl/05_0003_ddl_truncate_table.sql
> 
> What's the name rule ? I saw two 05_0000 sql file and none 05_0002 sql file

`05_0003` is the ddl/table catagory, we can name it to:
`05_0003_ddl_rename_table.sql`

**Comment by BohuTANG** [comment]
```
--- /home/runner/work/databend/databend/tests/suites/0_stateless/05_ddl/05_0003_ddl_rename_table.result	2022-03-15 03:56:35.945918604 +0000
+++ /home/runner/work/databend/databend/tests/suites/0_stateless/05_ddl/05_0003_ddl_rename_table.stdout	2022-03-15 03:58:22.258501441 +0000
@@ -1,3 +1,7 @@
+ERROR 1105 (HY000) at line 7: Code: 2302, displayText = table exists: t1.
+ERROR 1105 (HY000) at line 8: Code: 1054, displayText = Expected server error code: 1025 but got: Ok..
+1	NULL
+2	3
+1
 1
 1
-1
\ No newline at end of file
```

**Comment by kevinw66** [comment]
> ```
> --- /home/runner/work/databend/databend/tests/suites/0_stateless/05_ddl/05_0003_ddl_rename_table.result	2022-03-15 03:56:35.945918604 +0000
> +++ /home/runner/work/databend/databend/tests/suites/0_stateless/05_ddl/05_0003_ddl_rename_table.stdout	2022-03-15 03:58:22.258501441 +0000
> @@ -1,3 +1,7 @@
> +ERROR 1105 (HY000) at line 7: Code: 2302, displayText = table exists: t1.
> +ERROR 1105 (HY000) at line 8: Code: 1054, displayText = Expected server error code: 1025 but got: Ok..
> +1	NULL
> +2	3
> +1
>  1
>  1
> -1
> \ No newline at end of file
> ```

Is this because `02_0045_function_in.sql` create table `t1` but forgot to drop it?

**Comment by BohuTANG** [comment]
> > ```
> > --- /home/runner/work/databend/databend/tests/suites/0_stateless/05_ddl/05_0003_ddl_rename_table.result	2022-03-15 03:56:35.945918604 +0000
> > +++ /home/runner/work/databend/databend/tests/suites/0_stateless/05_ddl/05_0003_ddl_rename_table.stdout	2022-03-15 03:58:22.258501441 +0000
> > @@ -1,3 +1,7 @@
> > +ERROR 1105 (HY000) at line 7: Code: 2302, displayText = table exists: t1.
> > +ERROR 1105 (HY000) at line 8: Code: 1054, displayText = Expected server error code: 1025 but got: Ok..
> > +1	NULL
> > +2	3
> > +1
> >  1
> >  1
> > -1
> > \ No newline at end of file
> > ```
> 
> Is this because `02_0045_function_in.sql` create table `t1` but forgot to drop it?

Use another table name here?

**Comment by BohuTANG** [comment]
```
05_0003_ddl_rename_table:                                               [ FAIL ] - result differs with:
--- /Users/runner/work/databend/databend/tests/suites/0_stateless/05_ddl/05_0003_ddl_rename_table.result	2022-03-15 06:07:26.000000000 +0000
+++ /Users/runner/work/databend/databend/tests/suites/0_stateless/05_ddl/05_0003_ddl_rename_table.stdout	2022-03-15 06:10:03.000000000 +0000
@@ -1,3 +1,3 @@
 1
 1
-1
\ No newline at end of file
+1

```

It seems we should append a newline to the end of the result file.

**Review by Junnplus — APPROVED** [review]
Thanks!

LGTM on green.

**Pull request merged**